typings/index.d.ts

```diff
@@ -162,24 +162,20 @@ declare namespace commander {
      *
      *     // optional argument
      *     program.option('-c, --cheese [type]', 'add cheese [marble]');
-     *
-     * @param {string} flags
-     * @param {string} [description]
-     * @param {((arg1: any, arg2: any) => void) | RegExp} [fn] function or default
-     * @param {*} [defaultValue]
-     * @returns {Command} for chaining
      */
-    option(flags: string, description?: string, fn?: ((arg1: any, arg2: any) => void) | RegExp, defaultValue?: any): Command;
-    option(flags: string, description?: string, defaultValue?: any): Command;
+    option(flags: string, description?: string, defaultValue?: string): Command;
```

Comment:
The boolean needs to be added to `option(flags: string, description?: string, defaultValue?: string)` not to the regexp overload.

See calling example in tests: https://github.com/tj/commander.js/blob/b25cc3abf209a285576b4d117d6dbff54a2a6100/tests/options.bool.combo.test.js#L49

Reply:
As in: https://github.com/tj/commander.js/pull/1119#discussion_r361771928

Reply:
oups, sorry.